wpilibj/src/main/java/edu/wpi/first/wpilibj/simulation/SingleJointedArmSim.java

```diff
@@ -251,7 +251,7 @@ public double getVelocityRadPerSec() {
   public double getCurrentDrawAmps() {
     // Reductions are greater than 1, so a reduction of 10:1 would mean the motor is
     // spinning 10x faster than the output
-    var motorVelocity = m_x.get(1, 0) * m_gearing;
+    var motorVelocity = getVelocityRadPerSec() * m_gearing;
```

Comment:
This should still be using the state instead of the output.

Reply:
I'll also make these changes in DCMotorSim and ElevatorSim

Reply:
`  public LinearSystem<States, Inputs, ?> slice(int... outputIndices) {
    for (int index = 0; index < outputIndices.length; ++index) {
      if (outputIndices[index] < 0 || outputIndices[index] > m_C.getNumRows()) {
        throw new IllegalArgumentException(
            "Not all elements of outputIndices are in range of possible outputs. This is usually due to model implementation errors.");
      }
    }
    List<Integer> outputIndicesList = Arrays.stream(outputIndices).boxed().collect(Collectors.toList());
    Collections.sort(outputIndicesList);
    
    SimpleMatrix new_C_Storage = new SimpleMatrix(outputIndices.length, m_C.getNumCols());
    int row = 0;
    for(var index : outputIndicesList){
      var current_row_data = m_C.extractRowVector(index).getData();
      new_C_Storage.setRow(row, 0, current_row_data);
      row++;
    }

    Matrix<?, States> new_C = new Matrix<>(new_C_Storage);

    SimpleMatrix new_D_Storage = new SimpleMatrix(outputIndices.length, m_D.getNumCols());
    row = 0;
    for(var index : outputIndicesList){
      var current_row_data = m_D.extractRowVector(index).getData();
      new_D_Storage.setRow(row, 0, current_row_data);
      row++;
    }

    Matrix<?, Inputs> new_D = new Matrix<>(new_D_Storage);    

    LinearSystem<States, Inputs, ?> linearSystemSlice =  new LinearSystem<>(m_A, m_B, new_C, new_D);
    return linearSystemSlice;`

This is what I've got so far but I'm having trouble with types formulating linearSystemSlice near the bottom